shallow_backup.py

```diff
@@ -179,27 +186,69 @@ def backup_dotfiles(backup_path):
 
 	# Sublime Text Configs
 	if os.path.isdir("/Users/alichtman/Library/Application Support/Sublime\ Text\ 2"):
-		dotfolders_mp_in.append((os.path.join(home_path, "Library/Application\ Support/Sublime\ Text\ 2/Packages/User"), backup_path))
+		dotfolders_mp_in.append(
+			(os.path.join(home_path, "Library/Application\ Support/Sublime\ Text\ 2/Packages/User"), backup_path))
 
 	if os.path.isdir("/Users/alichtman/Library/Application Support/Sublime\ Text\ 3"):
-		dotfolders_mp_in.append((os.path.join(home_path, "Library/Application\ Support/Sublime\ Text\ 3/Packages/User"), backup_path))
+		dotfolders_mp_in.append(
+			(os.path.join(home_path, "Library/Application\ Support/Sublime\ Text\ 3/Packages/User"), backup_path))
 
 	# pprint(dotfiles_mp_in)
 	# pprint(dotfolders_mp_in)
 
 	# Multiprocessing
 	with mp.Pool(mp.cpu_count()):
 
-		print(Fore.BLUE + Style.BRIGHT + "Backing up dotfolders..." + Style.RESET_ALL)
+		print(Fore.BLUE + Style.BRIGHT +
+			  "Backing up dotfolders..." + Style.RESET_ALL)
 		for x in dotfolders_mp_in:
 			x = list(x)
 			mp.Process(target=copy_dotfolder, args=(x[0], x[1],)).start()
 
 	with mp.Pool(mp.cpu_count()):
-		print(Fore.BLUE + Style.BRIGHT + "Backing up dotfiles..." + Style.RESET_ALL)
+		print(Fore.BLUE + Style.BRIGHT +
+			  "Backing up dotfiles..." + Style.RESET_ALL)
 		for x in dotfiles_mp_in:
 			x = list(x)
-			mp.Process(target=copy_dotfile, args=(x[0], x[1],)).start()
+			mp.Process(target=_copy_file, args=(x[0], x[1],)).start()
+
+
+def backup_configs(backup_path):
+	make_dir_warn_overwrite(backup_path)
+
+	def home_prefix(path):
+		return os.path.join(os.path.expanduser('~'), path)
+
+	configs_dir_mapping = {"Library/Application Support/Sublime Text 2/Packages/User/": "sublime_2",
+						   "Library/Application Support/Sublime Text 3/Packages/User/": "sublime_3", }
+	plist_files = ["Library/Preferences/com.apple.Terminal.plist"]
+
+	# backup config dirs in backup_path/configs/<target>/
+	for config, target in configs_dir_mapping.items():
+		if os.path.isdir(home_prefix(config)):
+			configs_backup_path = os.path.join(backup_path, target)
+			_mkdir_or_pass(configs_backup_path)
+			_copy_dir_content(home_prefix(config), configs_backup_path)
+
+	# backup plist files in backup_path/configs/plist/
+	plist_backup_path = os.path.join(backup_path, "plist")
+	_mkdir_or_pass(plist_backup_path)
+	for plist in plist_files:
+		if os.path.exists(home_prefix(plist)):
+			_copy_dir_content(home_prefix(plist), plist_backup_path)
+
+
+def _copy_dir_content(source, target):
```

Comment:
Mesh into the pre-existing `copy_dir` function. Discussion in comments of PR.

Reply:
I would keep both methods just for now, as `_copy_dir_content(source, target)` copies a source to a target, but the existing `copy_dotfolder` copies a dir to a backup_path, which depends on the value of dir, therefore it owns some more logic. Further explanations will follow in an extra comment of the PR.

Reply:
Both methods copy one directory to another path. We should be able to merge them together into one function that can serve all purposes (without complicating the logic too much further.)

I think all that would be needed is one extra `else` condition in the original copy_dir function.

Reply:
The thing is, I do also backup sublime files in the backup_configs method, which means the method falls into the sublime if case, although Im would like it to reach this extra `else` condition. I could add an extra flag to the methods signature to distinguish the use cases, but that is also not a great solution, but if you find that better we can go with that :) 

I still think this merge of the methods comes together with a refactoring of the backup_dotfiles method.

Reply:
All that `if "sublime"` logic does is get the name for the folder based on the sublime version. So it either calls the folder `Sublime 2` or `Sublime 3`. You can control where these folders are placed based on the `backup_path` you pass in to `copy_dotfolder`. 

For example, if you pass in `shallow-backup/configs`, then the sublime folder that will be created is `shallow-backup/configs/sublime_2/` or `shallow-backup/configs/sublime_3/`, but if you pass in `shallow-backup/packages/`, then you would get`shallow-backup/packages/sublime_2/` or `shallow-backup/packages/sublime_3/`.

So, it should work for whatever use case you're trying to use it for. Does that make sense?

Reply:
Alright, then how about I refactor the backup_dotfiles method? I think we can drop the logic inside the copy_dotfolder method and simplify it. I just need to find some time to refactor it, since next days will be difficult for me.

Reply:
Sounds good to me. I think the logic in the copy_dotfolder method should be simplified, and I'm cool with the refactoring you're suggesting.